### PR TITLE
Some improvements for `quick-start.rst`

### PR DESCRIPTION
**Pull request by ojeda**
Reported-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Miguel Ojeda <ojeda@kernel.org>

### CODE REVIEW

File: Documentation/rust/quick-start.rst [hunk 1]
```diff
@@ -6,6 +6,9 @@ Quick Start
 This document describes how to get started with kernel development in Rust.
 If you have worked previously with Rust, this will only take a moment.
 
+Please note that, at the moment, a very restricted subset of architectures
+is supported, see :ref:`Documentation/rust/arch-support.rst <_rust_arch_support>`.
```
Comment (alex) [inline Documentation/rust/quick-start.rst]:
```suggestion
is supported, see :doc:`/rust/arch-support.rst`.
```

Comment (ojeda) [reply Documentation/rust/quick-start.rst]: +1

**Review by alex — APPROVED** [review]

File: Documentation/rust/quick-start.rst (hunk 1 above)
Comment (ojeda) [reply Documentation/rust/quick-start.rst]: It is:

    :doc:`Documentation/rust/arch-support.rst`

no? Otherwise I get a warning on `htmldocs`.

Comment (alex) [reply Documentation/rust/quick-start.rst]: I don't think so, none of the other `:doc:` links I see in there have the `Documentation/` prefix.

Comment (ojeda) [reply Documentation/rust/quick-start.rst]: Ah, I saw the problem: one needs to remove the extension:

    :doc:`/rust/arch-support`.

**Review by alex — APPROVED** [review]

**Pull request merged**